Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/modules.xml	(date 1503757225000)
+++ .idea/modules.xml	(revision )
@@ -2,8 +2,8 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/Knocksense.iml" filepath="$PROJECT_DIR$/Knocksense.iml" />
       <module fileurl="file://$PROJECT_DIR$/app/app.iml" filepath="$PROJECT_DIR$/app/app.iml" />
+      <module fileurl="file://$PROJECT_DIR$/knocksense.iml" filepath="$PROJECT_DIR$/knocksense.iml" />
     </modules>
   </component>
 </project>
\ No newline at end of file
Index: app/src/main/java/com/example/abhishek/knocksense/ArticleRecyclerViewAdapter.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/abhishek/knocksense/ArticleRecyclerViewAdapter.java	(revision d521f9107f9456783b0c714ed4421a5a664892cb)
+++ app/src/main/java/com/example/abhishek/knocksense/ArticleRecyclerViewAdapter.java	(revision )
@@ -1,13 +1,19 @@
 package com.example.abhishek.knocksense;
 
+import android.content.Context;
 import android.support.v7.widget.RecyclerView;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
+import android.widget.Toast;
 
+import com.android.volley.toolbox.ImageLoader;
+import com.android.volley.toolbox.NetworkImageView;
 import com.example.abhishek.knocksense.HomeFragment.OnListFragmentInteractionListener;
 import com.example.abhishek.knocksense.components.Article;
+import com.example.abhishek.knocksense.components.GlobalLists;
+import com.example.abhishek.knocksense.components.VolleySingleton;
 
 import java.util.List;
 
@@ -21,6 +27,7 @@
     private final List<Article> mValues;
     private final OnListFragmentInteractionListener mListener;
 
+    private ImageLoader mImageLoader;
     public ArticleRecyclerViewAdapter(List<Article> items, OnListFragmentInteractionListener listener) {
         mValues = items;
         mListener = listener;
@@ -42,6 +49,7 @@
         holder.author.setText(mValues.get(position).getAuthor());
         holder.date.setText(mValues.get(position).getDate());
 //        holder.featuredImage.setImageUrl(mValues.get(position).getFeaturedImage(), imageLoader);
+        holder.loadImage(position);
 
         holder.mView.setOnClickListener(new View.OnClickListener() {
             @Override
@@ -71,7 +79,7 @@
         public final TextView title;
         public final TextView author;
         public final TextView date;
-//        public final NetworkImageView featuredImage;
+      public final NetworkImageView featuredImage;
 
         public Article mItem;
 
@@ -81,13 +89,24 @@
             title = (TextView) view.findViewById(R.id.article_item_row_title);
             author = (TextView) view.findViewById(R.id.article_item_row_author);
             date = (TextView) view.findViewById(R.id.article_item_row_date);
-//            featuredImage = (NetworkImageView)view.findViewById(R.id.featured_image);
+           featuredImage = (NetworkImageView)view.findViewById(R.id.featured_image);
 
         }
+        private void loadImage(int position){
+            String url = mValues.get(position).getFeaturedImage();
 
+            mImageLoader = VolleySingleton.getInstance(GlobalLists.getContext())
+                    .getImageLoader();
+            mImageLoader.get(url, ImageLoader.getImageListener(featuredImage,
+                    R.drawable.ic_menu_camera, android.R.drawable
+                            .ic_dialog_alert));
+            featuredImage.setImageUrl(url, mImageLoader);
+        }
         @Override
         public String toString() {
             return super.toString() + " '" + author.getText() + "'";
         }
+
+
     }
 }
Index: app/src/main/java/com/example/abhishek/knocksense/components/VolleySingleton.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/com/example/abhishek/knocksense/components/VolleySingleton.java	(revision d521f9107f9456783b0c714ed4421a5a664892cb)
+++ app/src/main/java/com/example/abhishek/knocksense/components/VolleySingleton.java	(revision )
@@ -4,8 +4,13 @@
 import android.graphics.Bitmap;
 import android.support.v4.util.LruCache;
 
+import com.android.volley.Cache;
+import com.android.volley.Network;
 import com.android.volley.Request;
 import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.BasicNetwork;
+import com.android.volley.toolbox.DiskBasedCache;
+import com.android.volley.toolbox.HurlStack;
 import com.android.volley.toolbox.ImageLoader;
 import com.android.volley.toolbox.Volley;
 
@@ -51,10 +56,16 @@
         if (mRequestQueue == null) {
             // getApplicationContext() is key, it keeps you from leaking the
             // Activity or BroadcastReceiver if someone passes one in.
+            Cache cache = new DiskBasedCache(mCtx.getCacheDir(), 10 * 1024 * 1024);
+            Network network = new BasicNetwork(new HurlStack());
+            mRequestQueue = new RequestQueue(cache, network);
+            mRequestQueue.start();
+
             mRequestQueue = Volley.newRequestQueue(mCtx.getApplicationContext());
         }
         return mRequestQueue;
     }
+
 
     public <T> void addToRequestQueue(Request<T> req) {
         getRequestQueue().add(req);
Index: app/src/main/res/layout/content_main.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/res/layout/content_main.xml	(revision d521f9107f9456783b0c714ed4421a5a664892cb)
+++ app/src/main/res/layout/content_main.xml	(revision )
@@ -27,4 +27,5 @@
 
     </android.support.v4.view.ViewPager>
 
+
 </LinearLayout>
